recipes/h5pp/all/conanfile.py

```diff
@@ -30,9 +39,11 @@ def _compilers_minimum_version(self):
         }
 
     def requirements(self):
-        self.requires("eigen/3.4.0")
         self.requires("hdf5/1.12.1")
-        self.requires("spdlog/1.10.0")
+        if self.options.with_eigen:
+            self.requires("eigen/3.4.0")
+        if self.options.with_spdlog:
+            self.requires("spdlog/1.10.0")
```

Comment:
but is it really usable without eigen or spdlog before 1.10.0?

Reply:
Yes, older versions will work without eigen or spdlog. The h5pp headers will check for eigen and spdlog headers with the `#if __has_include` mechanism. I found this to be a bit fragile though so I've introduced these H5PP_USE_*** flags instead (to avoid collisions with system installed libraries for instance).

Reply:
I agree. I think this is why previous versions of h5pp would not give an option to disable these dependencies. Perhaps that should remain the case. I'm thinking something like this:

```python
    def requirements(self):
        self.requires("hdf5/1.12.1")
        if tools.Version(self.version) >= "1.10.0":
            if self.options.with_eigen:
                self.requires("eigen/3.4.0")
            if self.options.with_spdlog:
                self.requires("spdlog/1.10.0")
        else:
            # These dependencies are always required before h5pp 1.10.0:
            #   * h5pp < 1.10.0 includes any version of headers indiscriminately (e.g. system headers),
            #     and can't tell if the the corresponding library will be linked. This makes the,
            #     build and /link steps non-deterministic.
            #   * h5pp >= 1.10.0 fixes the issue with H5PP_USE_<LIB> preprocessor flags, to make sure
            #     that including the headers is intentional.
            self.requires("eigen/3.4.0")
            self.requires("spdlog/1.10.0")
            del self.options.with_eigen
            del self.options.with_spdlog
```
Shall I add this to the PR or do you know a better way?

Reply:
I think this is now solved by [9dffa21](https://github.com/conan-io/conan-center-index/commit/9dffa21a1e443450841819585e7d75247e8235e7)